numbers/mixtureModel.go

```diff
@@ -0,0 +1,256 @@
+package numbers
+
+import (
+	"math"
+	"math/rand"
+)
+
+// logProbEpsilon is the arbitrary set point for convergence. When the model likelihood is
+// increasing by < logProbEpsilon, we say the model has converged, possibly to a local minimum.
+const logProbEpsilon = 1e-08
+
+// MixtureModel holds data, results, and working memory for running the EM algorithm.
+type MixtureModel struct {
+	Data           []float64   // 1d data slice
+	K              int         // number of component distributions
```

Comment:
Is k a manual input, aka the user decides how many component distributions there are?

Reply:
That is correct. You need to have some prior information about how many components you are expecting. In my case, I am using it for genotyping and I assume that there are 2 copies of the autosomes or 1 of X/Y